src/NuGet.Core/NuGet.Packaging/PackageReaderBase.cs

```diff
@@ -568,7 +568,7 @@ protected static void ValidatePackageEntry(string normalizedDestination, string
 
         protected string NormalizeDirectoryPath(string path)
         {
-            if (!path.EndsWith(Path.DirectorySeparatorChar.ToString()))
+            if (!path.EndsWith(Path.DirectorySeparatorChar.ToString(CultureInfo.CurrentCulture)))
```

Comment:
Path separator chars (`\` or `/`) does not vary by culture

```suggestion
            if (!path.EndsWith(Path.DirectorySeparatorChar.ToString(CultureInfo.InvariantCulture)))
```